jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/support/UdpOutputWriterBuilder.java

```diff
@@ -69,11 +73,19 @@ private UdpOutputWriterBuilder(@Nonnull InetSocketAddress server, @Nonnull T tar
 						bufferSize,
 						charset,
 						flushStrategy))
-				.setExpiration(new DatagramChannelExpiration())
+				.setExpiration(createSocketExpiration())
 				.setSize(poolSize);
 		return new BlazePool<>(config);
 	}
 
+	private Expiration<DatagramChannelPoolable> createSocketExpiration() {
+		if (socketExpirationMs <= 0) {
+			return new DatagramChannelExpiration();
+		} else {
+			return new TimeExpiration<DatagramChannelPoolable>(socketExpirationMs, MILLISECONDS);
```

Comment:
Does a time based expiration really make sense for an UDP socket?

Reply:
I've added to both for consistency, but you're right, it doesn't make sense, removed it.